riker-macros/src/lib.rs

```diff
@@ -43,32 +46,60 @@ impl Parse for MsgTypes {
             types: vars
                 .into_iter()
                 .map(|t| MsgVariant {
-                    name: get_name(&t.path.segments),
+                    name: Ident::new(&get_path_name(&t), t.span()),
                     mtype: t,
                 })
                 .collect::<Vec<_>>(),
         })
     }
 }
 
-fn get_name(segments: &Punctuated<PathSegment, Colon2>) -> Ident {
-    let vname = segments
-        .iter()
-        .map(|seg| {
-            let ident = format!("{}", seg.ident);
-            ident
-                .split('_')
-                .map(|s| {
-                    let mut s = s.to_string();
-                    if let Some(c) = s.get_mut(0..1) {
-                        c.make_ascii_uppercase();
-                    }
-                    s
-                })
-                .collect::<String>()
-        })
-        .collect::<String>();
-    syn::Ident::new(&vname, segments.span())
+fn get_type_name(ty: &Type) -> String {
+    match ty {
+        Type::Path(path) => get_path_name(path),
+        Type::Array(TypeArray { elem, .. })
+        | Type::Group(TypeGroup { elem, .. })
+        | Type::Paren(TypeParen { elem, .. })
+        | Type::Ptr(TypePtr { elem, .. })
+        | Type::Reference(TypeReference { elem, .. })
+        | Type::Slice(TypeSlice { elem, .. }) => get_type_name(elem.as_ref()),
+        Type::Tuple(TypeTuple { elems, .. }) => {
+            elems.iter().fold(String::new(), |mut acc, curr| {
+                acc.push_str(&get_type_name(&curr));
+                acc
+            })
+        }
+        _ => String::new(),
```

Comment:
The draft looks really good, but it might be better to use unreachable for the default case in the match statement instead of an empty string. Using an empty string will produce duplicated idents if the left most part(s) get matched by the _ and the rest is the same for two types.

And I'd rather have it abort compilation here, than maybe do something unexpected because of two names being equal on accident.